### PR TITLE
Add longer descriptions for `fuelCycle` impls

### PR DESCRIPTION
**Pull request by keckler**
## What is the change?

Just adding some details to impl tags in the `fuelCycle` package.

## Why is the change being made?

Quality.

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [ ] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] If any [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were affected, mention it in the [release notes](https://terrapower.github.io/armi/release/index.html).
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/physics/fuelCycle/fuelHandlerInterface.py [hunk 1]
```diff
@@ -36,6 +36,23 @@ class FuelHandlerInterface(interfaces.Interface):
...
 
+        This interface allows for a user to define custom shuffle logic that
+        modifies to the core model. Being based on the :py:class:`~armi.interfaces.Interface`
+        class, it has direct access to the current core model.
+
+        User logic is able to be executed from within the
+        :py:meth:`~armi.physics.fuelCycle.fuelHandlerInterface.FuelHandlerInterface.manageFuel` method,
+        which will use the :py:meth:`~armi.physics.fuelCycle.fuelHandlerFactory.fuelHandlerFactory`
+        to search for a Python file specified by the case settings
+        ``fuelHandlerName`` and ``shuffleLogic``. If it exists, the user-specified
```
Comment (bsculac) [inline armi/physics/fuelCycle/fuelHandlerInterface.py:47]: The python file in question can be defined in two separate settings?

File: armi/physics/fuelCycle/fuelHandlers.py [hunk 2]
```diff
@@ -727,22 +727,38 @@ def swapAssemblies(self, a1, a2):
...
             :id: I_ARMI_SHUFFLE_STATIONARY0
             :implements: R_ARMI_SHUFFLE_STATIONARY
 
+            Before assemblies are moved,
+            calls to the ``_transferStationaryBlocks`` class method to
+            check if there are any block types specified by the user as stationary
+            via the ``stationaryBlockFlags`` case setting. Using these flags, gather
+            blocks from each assembly which should remain stationary and check
+            to make sure that both assemblies have the same number and same
+            height of stationary blocks. If not, return an error.
```
Comment (bsculac) [inline armi/physics/fuelCycle/fuelHandlers.py:745]:
```suggestion
            Before assemblies are moved,
            the ``_transferStationaryBlocks`` class method is called to
            check if there are any block types specified by the user as stationary
            via the ``stationaryBlockFlags`` case setting. Using these flags, blocks
            are gathered from each assembly which should remain stationary and 
            checked to make sure that both assemblies have the same number 
            and same height of stationary blocks. If not, return an error.
```

File: armi/physics/fuelCycle/fuelHandlers.py [hunk 3]
```diff
@@ -727,22 +727,38 @@ def swapAssemblies(self, a1, a2):
...
+
+            If all checks pass, use the :py:meth:`~armi.reactor.assemblies.Assembly.remove`
+            and :py:meth:`~armi.reactor.assemblies.Assembly.insert``
+            methods to swap the stationary blocks between the two assemblies.
```
Comment (bsculac) [inline armi/physics/fuelCycle/fuelHandlers.py:749]:
```suggestion
            If all checks pass, the :py:meth:`~armi.reactor.assemblies.Assembly.remove`
            and :py:meth:`~armi.reactor.assemblies.Assembly.insert``
            methods are used to swap the stationary blocks between the two assemblies.
```

File: armi/physics/fuelCycle/fuelHandlers.py [hunk 4]
```diff
@@ -835,10 +851,22 @@ def dischargeSwap(self, incoming, outgoing):
...
-        -----
-        The implementation for ``R_ARMI_SHUFFLE_STATIONARY`` occurs within
-        :py:meth:`<armi.physics.fuelCyle.fuelHandlers.FuelHandler._transferStationaryBlocks`.
+            Before assemblies are moved,
+            calls to the ``_transferStationaryBlocks`` class method to
+            check if there are any block types specified by the user as stationary
+            via the ``stationaryBlockFlags`` case setting. Using these flags, gather
+            blocks from each assembly which should remain stationary and check
+            to make sure that both assemblies have the same number and same
+            height of stationary blocks. If not, return an error.
```
Comment (bsculac) [inline armi/physics/fuelCycle/fuelHandlers.py:860]:
```suggestion
            Before assemblies are moved,
            the ``_transferStationaryBlocks`` class method is called to
            check if there are any block types specified by the user as stationary
            via the ``stationaryBlockFlags`` case setting. Using these flags, blocks
            are gathered from each assembly which should remain stationary and 
            checked to make sure that both assemblies have the same number 
            and same height of stationary blocks. If not, return an error.
```

File: armi/physics/fuelCycle/fuelHandlers.py [hunk 5]
```diff
@@ -835,10 +851,22 @@ def dischargeSwap(self, incoming, outgoing):
...
+
+            If all checks pass, use the :py:meth`~armi.reactor.assemblies.Assembly.remove`
+            and :py:meth:`~armi.reactor.assemblies.Assembly.insert``
+            methods to swap the stationary blocks between the two assemblies.
```
Comment (bsculac) [inline armi/physics/fuelCycle/fuelHandlers.py:864]:
```suggestion
            If all checks pass, the :py:meth:`~armi.reactor.assemblies.Assembly.remove`
            and :py:meth:`~armi.reactor.assemblies.Assembly.insert``
            methods are used to swap the stationary blocks between the two assemblies.
```

File: armi/physics/fuelCycle/fuelHandlers.py [hunk 6]
```diff
@@ -835,10 +851,22 @@ def dischargeSwap(self, incoming, outgoing):
             :id: I_ARMI_SHUFFLE_STATIONARY1
             :implements: R_ARMI_SHUFFLE_STATIONARY
```
Comment (bsculac) [inline armi/physics/fuelCycle/fuelHandlers.py:853]:
```suggestion
            :implements: R_ARMI_SHUFFLE_STATIONARY

            This is the implementation of the stationary requirement for the 
            discharge swap.
```

File: armi/physics/fuelCycle/fuelHandlers.py [hunk 7]
```diff
@@ -727,22 +727,38 @@ def swapAssemblies(self, a1, a2):
...
             :implements: R_ARMI_SHUFFLE_MOVE
 
+            For the two assemblies that are passed in, call to their :py:meth:`~armi.reactor.assemblies.Assembly.moveTo`
+            methods to transfer their underlying ``spatialLocator`` attributes to
+            each other. This will also update the ``childrenByLocator`` list on the
+            core as well as the assembly parameters ``numMoves`` and ``daysSinceLastMove``.
+
         .. impl:: User-specified blocks can be left in place and not moved.
             :id: I_ARMI_SHUFFLE_STATIONARY0
             :implements: R_ARMI_SHUFFLE_STATIONARY
```
Comment (bsculac) [inline armi/physics/fuelCycle/fuelHandlers.py:738]:
```suggestion
            :implements: R_ARMI_SHUFFLE_STATIONARY

            This is the implementation of the stationary requirement for the 
            within-core swap.
```

File: armi/physics/fuelCycle/fuelHandlerInterface.py [hunk 8]
```diff
@@ -36,6 +36,23 @@ class FuelHandlerInterface(interfaces.Interface):
         :id: I_ARMI_SHUFFLE
         :implements: R_ARMI_SHUFFLE
 
+        This interface allows for a user to define custom shuffle logic that
+        modifies to the core model. Being based on the :py:class:`~armi.interfaces.Interface`
+        class, it has direct access to the current core model.
```
Comment (john-science) [inline armi/physics/fuelCycle/fuelHandlerInterface.py:41]:
```suggestion
        This :py:class:`~armi.interfaces.Interface` allows a user to define custom
        shuffle logic that modifies to the core model.
```

Comment (john-science) [reply armi/physics/fuelCycle/fuelHandlerInterface.py:41]: I just thought the two sentences could be combined into one.  I'm not married to it, and I won't die on this hill.

File: armi/physics/fuelCycle/fuelHandlers.py (hunk 7 above)
Comment (john-science) [reply armi/physics/fuelCycle/fuelHandlers.py:738]: I would argue this text is so central, it should be covered by the "title" of the implementation (that first line with the `.. impl::` tag). So, perhaps this sentence is a hint that you want tweak the title?

File: armi/physics/fuelCycle/fuelHandlers.py (hunk 6 above)
Comment (john-science) [reply armi/physics/fuelCycle/fuelHandlers.py:853]: As I said above. I think the correct place for this sort of idea would be the title of the `impl` tag.

File: armi/physics/fuelCycle/fuelHandlerInterface.py (hunk 1 above)
Comment (keckler) [reply armi/physics/fuelCycle/fuelHandlerInterface.py]: Good call

https://github.com/terrapower/armi/pull/1589/commits/11d2c6ab0714ea8d436189aef574a11baff9f4ee

File: armi/physics/fuelCycle/fuelHandlers.py (hunk 7 above)
Comment (keckler) [reply armi/physics/fuelCycle/fuelHandlers.py:738]: https://github.com/terrapower/armi/pull/1589/commits/39d55597b00ba43df7be17ad69db9aedcdc0889d

File: armi/physics/fuelCycle/fuelHandlers.py (hunk 6 above)
Comment (keckler) [reply armi/physics/fuelCycle/fuelHandlers.py:853]: https://github.com/terrapower/armi/pull/1589/commits/39d55597b00ba43df7be17ad69db9aedcdc0889d

**Review by bsculac — APPROVED** [review]

**Pull request merged**